linera-views/src/reentrant_collection_view.rs

```diff
@@ -178,48 +178,76 @@ where
     ViewError: From<C::Error>,
     W: View<C> + Send + Sync,
 {
+    /// Reads the view and if missing returns the default view
     async fn wrapped_view(
         context: &C,
-        clear: bool,
+        needs_clear: bool,
         short_key: &[u8],
     ) -> Result<Arc<RwLock<W>>, ViewError> {
         let key = context.base_tag_index(KeyTag::Subview as u8, short_key);
         let context = context.clone_with_base_key(key);
         // Obtain a view and set its pending state to the default (e.g. empty) state
         let mut view = W::load(context).await?;
-        if clear {
+        if needs_clear {
             view.clear();
         }
         Ok(Arc::new(RwLock::new(view)))
     }
 
-    async fn try_load_view(
+    /// Reads the view if existing, otherwise report an error.
+    async fn checked_wrapped_view(
         context: &C,
-        updates: &mut BTreeMap<Vec<u8>, Update<Arc<RwLock<W>>>>,
-        was_cleared: bool,
+        needs_clear: bool,
         short_key: &[u8],
-    ) -> Result<Arc<RwLock<W>>, ViewError> {
-        use btree_map::Entry::*;
+    ) -> Result<Option<Arc<RwLock<W>>>, ViewError> {
+        let key_index = context.base_tag_index(KeyTag::Index as u8, short_key);
+        if needs_clear || !context.contains_key(&key_index).await? {
```

Comment:
When `needs_clear` is true, we can return `None` directly without computing `key_index`.

Reply:
This entire function could also be inlined (once) afaict.